src/main/java/gg/projecteden/nexus/features/shops/ResourceWorldCommand.java

```diff
@@ -0,0 +1,491 @@
+		send(PREFIX + "Added &e" + amount + " &3locations to logger, new size: &e" + getLogger(world()).size());
+	}
+
+	@NotNull
+	private Location getRandomLocation() {
+		final int x = RandomUtils.randomInt(-RADIUS, RADIUS);
+		final int y = RandomUtils.randomInt(-64, 319);
+		final int z = RandomUtils.randomInt(-RADIUS, RADIUS);
+		return new Location(world(), x, y, z);
+	}
+
+	@EventHandler
+	public void onEnterResourceWorld(PlayerTeleportEvent event) {
+		Player player = event.getPlayer();
+
+		if (!isResourceWorld(event.getTo()))
+			return;
+
+		if (isResourceWorld(event.getFrom()))
+			return;
+
+		if (Rank.of(player).isStaff())
+			return;
+
+		if (event.isCancelled())
+			return;
+
+		PlayerUtils.send(player, " &4Warning |");
+		PlayerUtils.send(player, " &4Warning | &cYou are entering the resource world!");
+		PlayerUtils.send(player, " &4Warning | &cThis world is regenerated on the &c&lfirst of every month&c,");
+		PlayerUtils.send(player, " &4Warning | &cso don't leave your stuff here or you will lose it!");
+		PlayerUtils.send(player, " &4Warning |");
+		PlayerUtils.send(player, " &4Warning | &cThe darkness is dangerous in this world");
+		PlayerUtils.send(player, " &4Warning |");
+	}
+
+	@EventHandler
+	public void onHomeBlockPlace(BlockPlaceEvent event) {
+		if (!isResourceWorld(event.getPlayer())) return;
+
+		List<Material> materials = new ArrayList<>(Arrays.asList(Material.CHEST, Material.TRAPPED_CHEST, Material.FURNACE, Material.BARREL));
+		materials.addAll(MaterialTag.WOODEN_DOORS.getValues());
+		if (!materials.contains(event.getBlockPlaced().getType()))
+			return;
+
+		Tip tip = new TipService().get(event.getPlayer());
+		if (tip.show(TipType.RESOURCE_WORLD_STORAGE))
+			PlayerUtils.send(event.getPlayer(), " &4Warning: &cYou are currently building in the resource world! " +
+				"This world is regenerated on the &c&lfirst of every month, &cso don't leave your stuff here or you will lose it!");
+	}
+
+	@EventHandler
+	public void onBlockPlace(BlockPlaceEvent event) {
+		final Block block = event.getBlock();
+		final World world = block.getWorld();
+		if (!isResourceWorld(world))
+			return;
+
+		getLogger(world).add(block.getLocation());
+		save(world);
+	}
+
+	@EventHandler
+	public void onBlockBreak(BlockBreakEvent event) {
+		final Block block = event.getBlock();
+		final World world = block.getWorld();
+		if (!isResourceWorld(world))
+			return;
+
+		Tasks.wait(2, () -> {
+			if (block.getType() != Material.AIR)
+				return;
+
+			getLogger(world).remove(block.getLocation());
+			save(world);
+		});
+	}
+
+	@EventHandler
+	public void onBlockPistonExtend(BlockPistonExtendEvent event) {
+		handlePiston(event.getBlock(), event.getBlocks(), event.getDirection());
+	}
+
+	@EventHandler
+	public void onBlockPistonExtend(BlockPistonRetractEvent event) {
+		handlePiston(event.getBlock(), event.getBlocks(), event.getDirection());
+	}
+
+	private void handlePiston(Block eventBlock, List<Block> blocks, BlockFace direction) {
```

Comment:
has this been tested with slime/honey blocks?

Reply:
Yes